client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx

```diff
@@ -134,141 +170,165 @@ const Review: React.FC<IProps> = ({ prevStage, locked, refresh }: IProps) => {
+                </a>
+              </td>
             </tr>
-          ))}
-        </tbody>
-      </ContestsTable>
-      <ContestsTable>
-        <thead>
-          <tr>
-            <th>Opportunistic Contests</th>
-            <th>Jurisdictions</th>
-          </tr>
-        </thead>
-        <tbody>
-          {opportunisticContests.map((c: IContest) => (
-            <tr key={c.id}>
-              <td>{c.name}</td>
-              <td>{c.jurisdictionIds.join(', ')}</td>
+            <tr>
+              <td>Audit Board Data Entry:</td>
+              <td>{online ? 'Online' : 'Offline'}</td>
             </tr>
-          ))}
-        </tbody>
-      </ContestsTable>
-      <H4>Sample Size Options</H4>
+          </tbody>
+        </SettingsTable>
+        <ContestsTable>
+          <thead>
+            <tr>
+              <th>Target Contests</th>
+              <th>Jurisdictions</th>
+            </tr>
+          </thead>
+          <tbody>
+            {targetedContests.map((c: IContest) => (
+              <tr key={c.id}>
+                <td>{c.name}</td>
+                <td>{c.jurisdictionIds.join(', ')}</td>
+              </tr>
+            ))}
+          </tbody>
+        </ContestsTable>
+        <ContestsTable>
+          <thead>
+            <tr>
+              <th>Opportunistic Contests</th>
+              <th>Jurisdictions</th>
+            </tr>
+          </thead>
+          <tbody>
+            {opportunisticContests.map((c: IContest) => (
+              <tr key={c.id}>
+                <td>{c.name}</td>
+                <td>{c.jurisdictionIds.join(', ')}</td>
+              </tr>
+            ))}
+          </tbody>
+        </ContestsTable>
+      </ElevatedCard>
+      <br />
+      <H4>Estimated Sample Sizes</H4>
       <Formik
         initialValues={{
-          sampleSize: sampleSizeOptions.length ? sampleSizeOptions[0].size : '',
-          customSampleSize: '',
+          sampleSizes: initialValues,
+          customSampleSizes: initialCustomValues,
         }}
         enableReinitialize
-        onSubmit={v => {
-          if (v.sampleSize === 'custom') {
-            setSampleSize(v.customSampleSize)
-          } else {
-            setSampleSize(v.sampleSize)
-          }
+        onSubmit={values => {
+          setSampleSizes(
+            values.sampleSizes.reduce((a, s, i) => {
+              const contestId = Object.keys(s)[0]
+              if (s[contestId] === 'custom') {
+                return {
+                  ...a,
+                  [contestId]: values.customSampleSizes[i][contestId],
+                }
+              }
+              return { ...a, [contestId]: s[contestId] }
+            }, {})
+          )
           setIsConfirmDialogOpen(true)
         }}
       >
         {({
           values,
           handleSubmit,
           setFieldValue,
-        }: FormikProps<{ sampleSize: string }>) => (
+        }: FormikProps<{ sampleSizes: { [key: string]: string }[] }>) => (
```

Comment:
why a list of objects? i think life might be simpler with a mapping from contest id to sample size, so `{ [contestId: string]: number }`

Reply:
@jonahkagan My concern was that we don't know the keys (the contest ids being the keys), and so having it in an array would make it more palatable for the form. But I ended up having to do weird stuff to get at it anyway, so I don't think it ended up simplifying it in the end. I'm going to try if it will work with just the object.